### PR TITLE
updated Routes module to latest generator

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | #3118
| Refs tickets      | -
| License           | MIT
| Changelog updated | no

### CODE REVIEW

File: src/system/RoutesModule/Entity/RouteEntity.php [hunk 1]
```diff
@@ -59,18 +59,20 @@ public function __construct()
...
-            return '/' . $modinfo['url'] . $this->path;
+        if (!\ServiceUtil::hasContainer()) {
+            \ServiceUtil::setContainer($container);
         }
 
-        return $this->path;
+        $modinfo = \ModUtil::getInfoFromName($bundle);
+
+        // return path prepended with bundle prefix
+        return '/' . $modinfo['url'] . $this->path;
```
Comment (craigh) [inline src/system/RoutesModule/Entity/RouteEntity.php]: ServiceUtil & ModUtil needs to be removed. Get the Bundle's MetaData and get the url from that.

File: src/system/RoutesModule/Form/Handler/Common/Base/AbstractEditHandler.php [hunk 2]
```diff
@@ -194,38 +195,39 @@ public function processForm(array $templateParameters)
     {
         $this->templateParameters = $templateParameters;
         $this->templateParameters['inlineUsage'] = UserUtil::getTheme() == 'ZikulaPrinterTheme' ? true : false;
-
-
+    
+
```
Comment (craigh) [inline src/system/RoutesModule/Form/Handler/Common/Base/AbstractEditHandler.php]: can we get the generator to stop adding empty tabs?

File: src/system/RoutesModule/Form/Type/Base/AbstractRouteType.php [hunk 3]
```diff
@@ -276,27 +276,25 @@ public function addReturnControlField(FormBuilderInterface $builder, array $opti
...
-     * @param array                $options The options.
+     * @param FormBuilderInterface $builder The form builder
+     * @param array                $options The options
      */
     public function addSubmitButtons(FormBuilderInterface $builder, array $options)
     {
         foreach ($options['actions'] as $action) {
             $builder->add($action['id'], 'Symfony\Component\Form\Extension\Core\Type\SubmitType', [
-                'label' => $this->__(/** @ignore */$action['title']),
+                'label' => $this->__($action['title']),
```
Comment (craigh) [inline src/system/RoutesModule/Form/Type/Base/AbstractRouteType.php]: the `/** @ignore */` is required for the translation extractor to work properly. please return.

File: src/system/RoutesModule/Resources/views/includePdfHeader.html.twig [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html xml:lang="{{ app.request.locale }}" lang="{{ app.request.locale }}" dir="{{ langdirection() }}">
+<html xml:lang="{{ lang() }}" lang="{{ lang() }}" dir="{{ langdirection() }}">
```
Comment (craigh) [inline src/system/RoutesModule/Resources/views/includePdfHeader.html.twig]: `lang()` will be removed. please use `app.request.locale`

File: src/system/RoutesModule/Form/Handler/Common/Base/AbstractEditHandler.php (hunk 2 above)
Comment (Guite) [inline src/system/RoutesModule/Form/Handler/Common/Base/AbstractEditHandler.php]: Not until https://bugs.eclipse.org/bugs/show_bug.cgi?id=398172 is resolved.

File: src/system/RoutesModule/Form/Type/Base/AbstractRouteType.php (hunk 3 above)
Comment (Guite) [inline src/system/RoutesModule/Form/Type/Base/AbstractRouteType.php]: The root problem of that is that workflow file's attributes (in the workflow xml definition) can not be translated. Any idea how to improve that?

Comment (craigh) [inline src/system/RoutesModule/Form/Type/Base/AbstractRouteType.php]: ftr, the `@ignore` tag doesn't prevent translation, it only prevents the translator extractor from  extracting the string that follows it (e.g. `$action['title']` literal)

File: src/system/RoutesModule/Entity/RouteEntity.php (hunk 1 above)
Comment (Guite) [inline src/system/RoutesModule/Entity/RouteEntity.php]: Is there any example for that in the core?

Comment (craigh) [inline src/system/RoutesModule/Entity/RouteEntity.php]: you already have the bundle in line 67 above. so do, `$bundle->getMetaData()->getUrl();`

**Comment by craigh** [comment]
obviously, you also need to solve the travis build before merging this.

**Pull request merged**